test/alternator/test_manual_requests.py

```diff
@@ -338,3 +338,38 @@ def test_base64_malformed_cond_expr(dynamodb, test_table_b):
             "begins_with(:v, c)"]:
         r = scan_with_binary_data_in_cond_expr(dynamodb, test_table_b, exp, exp_attr)
         assert r.status_code == 400, "Failed on expression \"%s\"" % (exp)
+
+# Tests some invalid payloads (empty requests) to BatchWriteItem. Reproduces #23233
+# The test expects ValidationException or SerializationException returned.
+# The test specifically forbids 'assert' word in message to check it is not a fallback
+# from RAPIDJSON_ASSERT re-implementation added in util/rjson.hh, that prevents crashing
+# on mishandled requests, but produces unreadable message (see #23233).
+def test_batch_write_item_invalid_payload(dynamodb, test_table, test_table_s):
+    cases = ['', 'null', '""', '{}',
+             '{"RequestItems": null}',
+             '{"RequestItems": ""}',
+             '{"RequestItems": []}',
+             '{"RequestItems": {}}',
+             '{"RequestItems": {"__TABLE__": null}}',
+             '{"RequestItems": {"__TABLE__": []}}',
+             '{"RequestItems": {"__TABLE__": [null]}}',
+             '{"RequestItems": {"__TABLE__": [{}}]}}',
+             '{"RequestItems": {"__TABLE__": [{"PutRequest": null}]}}',
+             '{"RequestItems": {"__TABLE__": [{"PutRequest": {}}]}}',
+             '{"RequestItems": {"__TABLE__": [{"PutRequest": {"Item": null}}]}}',
+             '{"RequestItems": {"__TABLE__": [{"PutRequest": {"Item": {}}}]}}',
+             '{"RequestItems": {"__TABLE__": [{"DeleteRequest": null}]}}',
+             '{"RequestItems": {"__TABLE__": [{"DeleteRequest": {}}]}}',
+             '{"RequestItems": {"__TABLE__": [{"DeleteRequest": {"Key": null}}]}}',
+             '{"RequestItems": {"__TABLE__": [{"DeleteRequest": {"Key": {}}}]}}'
+             '{"RequestItems": {"__TABLE_S__": [{"PutRequest": {"Item": {"p": {"S": "hello"}}}}], "__TABLE__": []}}'
+            ]
+    for body in cases:
+        body = body.replace("__TABLE__", test_table.name).replace("__TABLE_S__", test_table_s.name)
+        req = get_signed_request(dynamodb, "BatchWriteItem", body)
+        response = requests.post(req.url, headers=req.headers, data=req.body, verify=False)
+        assert "assert" not in response.text, f"RAPIDJSON_ASSERT fallback message for payload: \'{body}\'"
```

Comment:
Nice, I think this is the first time I saw a valid use for the second parameter to assert in pytest, and a good solution to the debugging errors that @ScyllaPiotr worried about and the bazillion-test problem I was worried about. Kudos!